.circleci/config.yml

```diff
@@ -62,7 +62,11 @@ jobs:
           name: Deploy to S3 if tests pass and branch is Master
           command: |
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              aws s3 sync /tmp/reaction-docs ${API_DOC_BUCKET} --delete --region ${API_DOC_BUCKET_REGION}
+              if [[ "${API_DOC_BUCKET}" && "${API_DOC_BUCKET_REGION}" ]]; then
```

Comment:
I'm not certain this will ensure that `${API_DOC_BUCKET}` and `${API_DOC_BUCKET_REGION}` are set in bash.

https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

Reply:
so
```
if [ "${CIRCLE_BRANCH}" == "master" ] && [[ -v API_DOC_BUCKET ]] ;  then
```

Reply:
That `-f` in the second example checks if the variable resolves to a path to a file that exists.  

https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html

Side note on that... it would also require quotes because of the single brackets.  If it ever resolved to `undefined`, that becomes an invalid bash expression.  

So, when not set, this...

```sh
if [ -f $API_DOC_BUCKET ];
```

is evaluated as...

```sh
if [ -f ];
```

Which is invalid and would tank the script.

Anyway, for the sake of simplicity and flexibility, I usually prefer the double brackets with nothing more than the var names.  Quotes are optional and you can do any amount of conditions within one set of brackets.  That said, his original code is valid.

```bash
if [[ $API_DOC_BUCKET && $API_DOC_BUCKET_REGION ]]; then
```
You can give it a try by creating a test script and set/unset the values...

```bash
#!/bin/bash

MY_VAR="stuff"
MY_VAR2="things"

if [[ $MY_VAR && $MY_VAR2 ]]; then
  echo "Both vars are set!"
else
  echo "Nope! One or both vars isn't set"
fi
```

So anyway, I think we're good as far as logic goes. I guess my only nitpick is the message when the vars aren't set.  It could use a little more detail.

Maybe something like...

```sh
echo "S3 bucket configuration not found for jsdocs"
echo "Set API_DOC_BUCKET and API_DOC_BUCKET_REGION to build and deploy jsdocs to S3"
```

Reply:
That `-f` in the second example checks if the variable resolves to a path to a file that exists.  

https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html

Side note on that... it would also require quotes because of the single brackets.  If it ever resolved to `undefined`, that becomes an invalid bash expression.  

So, when not set, this...

```sh
if [ -f $API_DOC_BUCKET ];
```

is evaluated as...

```sh
if [ -f ];
```

Which is invalid and would tank the script.

Anyway, for the sake of simplicity and flexibility, I usually prefer the double brackets with nothing more than the var names.  Quotes are optional and you can do any amount of conditions within one set of brackets.  That said, his original code is valid.

```bash
if [[ $API_DOC_BUCKET && $API_DOC_BUCKET_REGION ]]; then
```
You can give it a try by creating a test script and set/unset the values...

```bash
#!/bin/bash

MY_VAR="stuff"
MY_VAR2="things"

if [[ $MY_VAR && $MY_VAR2 ]]; then
  echo "Both vars are set!"
else
  echo "Nope! One or both vars isn't set"
fi
```

So anyway, I think we're good as far as logic goes. I guess my only nitpick is the message when the vars aren't set.  It could use a little more detail.

Maybe something like...

```sh
echo "S3 bucket configuration not found for jsdocs"
echo "Set API_DOC_BUCKET and API_DOC_BUCKET_REGION to build and deploy jsdocs to S3"
```

Reply:
@jshimko Also I would like to say that the following scipt
```bash
#!/bin/bash

MY_VAR="stuff"
MY_VAR2="things"

if [[ $MY_VAR && $MY_VAR2 ]]; then
  echo "Both vars are set!"
else
  echo "Nope! One or both vars isn't set"
fi
```
will output `Nope! One or both vars isn't set`, which is what we need.

I will update the message.